File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        first_occurrence = self.findOccurrence(nums, target, first = True, last = False)
        last_occurrence = self.findOccurrence(nums, target, first = False, last = True)
        return [first_occurrence, last_occurrence]
        
    def findOccurrence(self, nums: List[int], target: int, first: bool, last: bool) -> int:
        start = 0
        end = len(nums) - 1
        occurrence = -1
        while start <= end:
            mid = start + (end - start) // 2
            if (target == nums[mid]):
                occurrence = mid
                if first:
                    end = mid - 1
                if last:
                    start = mid + 1
            elif (target < nums[mid]):
                end = mid - 1
            elif (target > nums[mid]):
                start = mid + 1
        return occurrence
                
        

File: 34-find-first-and-last-position-of-element-in-sorted-array/34-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int first_occurrence = occurrence(nums, target, true, false);
        int last_occurrence = occurrence(nums, target, false, true);
        int[] output = {first_occurrence, last_occurrence};
        return output;
        
    }
    public int occurrence(int[] nums, int target, boolean first, boolean last) {
        int start = 0;
        int end = nums.length - 1;
        int position = -1;
        while(start <= end) {
            int mid = start + (end - start) / 2;
            if (target == nums[mid]) {
                position = mid;
                if (first) {
                    end = mid - 1;
                }
                if (last) {
                    start = mid + 1;
                }
            }
            else if (target < nums[mid]) {
                end = mid - 1;
            }
            else if (target > nums[mid]) {
                start = mid + 1;
            }
        }
        return position;
    }
}